src/overlays/actors/ovl_Demo_Ik/z_demo_ik.c

```diff
@@ -9,27 +10,506 @@ void DemoIk_Destroy(Actor* thisx, GlobalContext* globalCtx);
+    }
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, phi_f0);
+    SkelAnime_Init(globalCtx, &this->skelAnime, skeleton, NULL, this->jointTable, this->morphTable, 2);
+    Animation_Change(&this->skelAnime, animation, 1.0f, 0.0f, Animation_GetLastFrame(animation), 2, 0.0f);
+}
+
+void func_8098393C(DemoIk* this) {
+    this->actionMode = 0;
+    this->drawMode = 0;
+    this->actor.shape.shadowAlpha = 0;
+}
+
+void func_8098394C(DemoIk* this, GlobalContext* globalCtx) {
+    DemoIk_EndMove(this);
+    DemoIk_MoveToStartPos(this, globalCtx, DemoIk_GetIndexFromParams(this->actor.params));
+    this->actionMode = 1;
+    this->drawMode = 1;
+    this->actor.shape.shadowAlpha = 0xFF;
+    this->skelAnime.curFrame = 0.0f;
+}
+
+void func_809839AC(DemoIk* this) {
+    this->actionMode = 2;
+    this->drawMode = 1;
+    this->actor.shape.shadowAlpha = 0xFF;
+    this->skelAnime.curFrame = 0.0f;
+}
+
+void func_809839D0(DemoIk* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* cue = DemoIk_GetCue(globalCtx, DemoIk_GetIndexFromParams(this->actor.params));
+
+    if (cue != NULL) {
+        s32 nextCsAction = cue->action;
+        s32 csAction = this->csAction;
+
+        if (nextCsAction != csAction) {
+            switch (nextCsAction) {
+                case 1:
+                    func_8098393C(this);
+                    break;
+                case 2:
+                    func_8098394C(this, globalCtx);
+                    break;
+                case 3:
+                    func_809839AC(this);
+                    break;
+                case 4:
+                    Actor_Kill(&this->actor);
+                    break;
+                case 5:
+                case 6:
+                    break;
+                default:
+                    // there is no such action
+                    osSyncPrintf("Demo_Ik_Check_DemoMode:そんな動作は無い!!!!!!!!\n");
+            }
+            this->csAction = nextCsAction;
+        }
+    }
+}
+
+void DemoIk_Type1Action0(DemoIk* this, GlobalContext* globalCtx) {
+    func_809839D0(this, globalCtx);
+}
+
+void DemoIk_Type1Action1(DemoIk* this, GlobalContext* globalCtx) {
+    DemoIk_BgCheck(this, globalCtx);
+    func_809839D0(this, globalCtx);
+}
+
+void DemoIk_Type1Action2(DemoIk* this, GlobalContext* globalCtx) {
+    DemoIk_UpdateSkelAnime(this);
+    DemoIk_Type1PlaySound(this);
+    DemoIk_SetMove(this, globalCtx);
+    DemoIk_BgCheck(this, globalCtx);
+    DemoIk_SpawnDeadDb(this, globalCtx);
+    func_809839D0(this, globalCtx);
+}
+
+void DemoIk_Type1PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+    DemoIk* this = THIS;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+
+    OPEN_DISPS(gfxCtx, "../z_demo_ik_inArmer.c", 385);
+    if (limbIndex == 1) {
+        switch (this->actor.params) {
+            case 0:
+                gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(gfxCtx, "../z_demo_ik_inArmer.c", 390), 2);
```

Comment:
The last argument in all the gSPMatrix in this file need to be fixed to use the usual 3 gbi defines